src/compiler/parser.ts

```diff
@@ -923,6 +937,57 @@ namespace ts {
             return token > SyntaxKind.LastReservedWord;
         }
 
+            // ERRORS
+            function invalidTypeReferenceOrTypePredicate(node: TypeReferenceNode | TypePredicateNode): void {
+                if (node.kind === SyntaxKind.TypePredicate) {

```

Comment:
This is an after-the-fact grammar check. Can you explain why you couldn't just make this part of a grammar check in `checker.ts`?